server/src/graphql/schema.graphql

```diff
@@ -11,81 +11,98 @@
 schema {
   query: Query
   mutation: Mutation
-  subscription: Subscription
+  # subscription: Subscription
 }
 
+# type Subscription {
+#   surveyUpdates (surveyId: Int!): Survey
+# }
+
 type Query {
-  # Returns the logged-in user making the query, or null if anonymous.
-  self: User
+  # Returns a Party if one exists with the partyName, and its optional password matches the optional partyPassword.
+  # Otherwise, null.
+  party (partyName: String!, partyPassword: String): Party
 
-  surveys: [Survey!]!
-  survey (surveyId: Int!): Survey
+  # Returns all Song sorted by song title.
+  songs: [Song!]!
 }
 
 type Mutation {
-  # Records an answer to a survey question presented in class.
-  answerSurvey (input: SurveyInput!): Boolean!
-
-  # Moves the survey to the next question (or starts it if it hasn't started). ADMIN only.
-  nextSurveyQuestion (surveyId: Int!): Survey
-}
-
-type Subscription {
-  surveyUpdates (surveyId: Int!): Survey
+  # Creates a Party with the optional partyPassword if one does not yet exist with the partyName.
+  # Returns that newly created Party, or null if a Party already exists with the partyName.
+  createParty (partyName: String!, partyPassword: String): Party
+
+  # Votes for a Song in the context of a Party.
+  # Returns the VotedSong if the voting was successful (partyId and songId must exist), or null if unsuccessful.
+  vote (partyId: Int!, songId: Int!): VotedSong
+
+  # Advances the CurrentSong of a Party to the next song, which is the VotedSong with the highest VotedSong.count
+  # out of all the VotedSong for the Party. If there is no VotedSong for the Party, the CurrentSong becomes null.
+  # Returns the Party if successful, or null if unsuccessful.
+  nextSong(partyId: Int!): Party
 }
 
-type User {
+type Party  {
   id: Int!
-  userType: UserType!
-  email: String!
+
+  # The name of this Party that was inputted by the party creator.
+  # Must be unique among all Party.
   name: String!
-}
 
-enum UserType { ADMIN, USER }
+  # The plaintext password for this Party, or null if the Party creator did not specify a password.
+  password: String
 
-type Survey {
-  id: Int!
+  # The time that the Party was last interacted with. ISO 8601 format for date and time in UTC.
+  latestTime: String!
```

Comment:
Why do we need to expose this variable? I can't think of a client-side use case.

Reply:
Will our background process be interacting directly with the database layer, or through the API? If the former, then yeah I suppose we won't need latestTime.

Reply:
Good question, I'm actually not sure. We can ask someone on Friday and remove this if its unneeded.
I don't know if we have to worry about security for this project, but it would be pretty insecure to allow unauthenticated API requests to delete parties.